src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -68,6 +70,17 @@ export class RundownTimelineService implements RundownService {
     return timeline
   }
 
+  private doInfinitePiecesMapsDiffer(first: Map<string, Piece>, second: Map<string, Piece>):boolean {
+    if ( first !== null && second != null){
+      if (first.size != second.size) return true
+      first.forEach((firstPiece, firstMapKey)=>{
+        const secondPiece: Piece | undefined = second.get(firstMapKey)
+        if (secondPiece !== undefined && secondPiece.id !== firstPiece.id) return true
+        else return true
+      })
+    }
+    return false
+  }
```

Comment:
From my understanding the return will only break out of the inner loops and still return false when we should return true inside the forEach.  So if i have two maps of the same size butwith different content, it would still return false. 
I came up with a suggestion that i think covers the scenarios, please do look it over and see if you agree :-) 

``` 
  private doInfinitePiecesMapsDiffer(first: Map<string, Piece>, second: Map<string, Piece>): boolean {
    if (first === null && second === null) {
      return false
    }

    if (first.size !== second.size) {
      return true
    }

    for (const [firstMapKey, firstPiece] of first) {
      const secondPiece: Piece | undefined = second.get(firstMapKey)
      if (secondPiece === undefined || secondPiece.id !== firstPiece.id) {
        return true
      }
    }
    return false
  }

Reply:
The type signature of the method does not take a null values, thus the null check can be removed. And instead of using the for-loop, you can you use `Array.some`. For better understanding what the maps are, they can be renamed to something like `firstInfinitePieceMap` and `secondInfinitePieceMap`. An example can be seen below:
```typescript
  private doInfinitePieceMapsDiffer(firstInfinitePieceMap: Map<string, Piece>, secondInfinitePieceMap: Map<string, Piece>): boolean {
    if (firstInfinitePieceMap.size !== secondInfinitePieceMap.size) {
      return true
    }
    return [...firstInfinitePieceMap.entries()].some(([layer, piece]) => secondInfinitePieceMap.get(layer)?.id !== piece.id)
  }
```